### PR TITLE
refactor(behavior_velocity): occlusion spot merge scene modules

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Related Issue(required)

#258

## Description(required)

This PR is  going to change
- merge scene modules into one.
- refactor for adding velocity smoother https://github.com/autowarefoundation/autoware.universe/pull/497
- move detection area creation to util
- replace function with tier4 autoware utils

pass_judge option is going to be enable for velocity smoother PR #497 currently "smooth_velocity" is not available
```
      detection_method: "occupancy_grid" # [-] candidate is "occupancy_grid" or "predicted_object"
      pass_judge: "current_velocity"       # [-] candidate is "current_velocity" or "smooth_velocity"
```

This PR is NOT going to change
- adding new features or adding new files
- parameter change

## Review Procedure(required)

- make sure behavior is the same as before. below option should be tested
```
      detection_method: "occupancy_grid" # [-] candidate is "occupancy_grid" or "predicted_object"
```
- test in experiment at kashiwanoha

with #497

https://user-images.githubusercontent.com/65527974/158551188-468312cc-6472-42c1-9a3e-a6e27d51578b.mp4

https://user-images.githubusercontent.com/65527974/158551202-2fae6da4-7c46-4346-b4e0-bc4861e372bd.mp4

## Related PR(optional)

#497

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by yukkysaito** [comment]
@taikitanaka3 Is the detection area supposed to use smoother results?

**Comment by taikitanaka3** [comment]
@yukkysaito 
create detection area polygon only converts velocity in path so it can convert any path point velocity such as 
- current velocity into path
- velocity of constant acceleration in path(not supported nor wont make this)
 in this PR current velocity is applied to path to make detection area.
However with PR #497 
- velocity of smoothed path consider external velocity limit 
is available as an option.

**Comment by yukkysaito** [comment]
> velocity is applied to path to make detection area.

In this PR,  are you using the maximum velocity that is in the path and v0,a0?

**Comment by taikitanaka3** [comment]
@yukkysaito 
in this PR there is no function change as described in PR so current velcity is only available
- current velocity into path(only consider v0)

File: planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp [hunk 1]
```diff
@@ -36,14 +36,31 @@ OcclusionSpotModuleManager::OcclusionSpotModuleManager(rclcpp::Node & node)
...
-  if (method == "occupancy_grid") {
-    pp.method = METHOD::OCCUPANCY_GRID;
-  } else if (method == "predicted_object") {
-    pp.method = METHOD::PREDICTED_OBJECT;
-  } else {
-    throw std::invalid_argument{"[behavior_velocity]: occlusion spot detection method is invalid"};
+  // for detection type
+  {
+    const std::string method = node.declare_parameter(ns + ".detection_method", "occupancy_grid");
+    if (method == "occupancy_grid") {  // module id 0
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply planning/behavior_velocity_planner/src/scene_module/occlusion_spot/manager.cpp:42]: If these methods integrate, you don't have to modify this.

**Review by yukkysaito — DISMISSED** [review]

**Comment by taikitanaka3** [comment]
image
make sure behavior is same as before(current velocity = rectangle polygon)

**Comment by taikitanaka3** [comment]
- [x] release note

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**